sound/soc/sof/ipc4-topology.c

```diff
@@ -1493,12 +1546,27 @@ static int sof_ipc4_route_setup(struct snd_sof_dev *sdev, struct snd_sof_route *
 	struct sof_ipc4_fw_module *sink_fw_module = sink_widget->module_info;
 	struct sof_ipc4_msg msg = {{ 0 }};
 	u32 header, extension;
-	int src_queue = 0;
-	int dst_queue = 0;
 	int ret;
 
-	dev_dbg(sdev->dev, "bind %s -> %s\n",
-		src_widget->widget->name, sink_widget->widget->name);
+	sroute->src_queue_id = sof_ipc4_get_queue_id(src_widget, SOF_ROUTE_BIND_SINK,
+						     sink_widget->widget->name);
+	if (sroute->src_queue_id < 0) {
+		dev_err(sdev->dev, "failed to get source queue id for %s\n",
+			src_widget->widget->name);
+		return sroute->src_queue_id;
+	}
+
+	sroute->dst_queue_id = sof_ipc4_get_queue_id(sink_widget, SOF_ROUTE_BIND_SOURCE,
+						     src_widget->widget->name);
+	if (sroute->dst_queue_id < 0) {
+		dev_err(sdev->dev, "failed to get sink queue id for%s\n",
+			sink_widget->widget->name);
+		return sroute->dst_queue_id;
```

Comment:
error handling doesn't release what was allocated above.

Are we missing a 
sof_ipc4_get_put_id(sroute->src_queue_id);

???